render_silhouette_regmarks.py

```diff
@@ -0,0 +1,132 @@
+REG_MARK_LINE_WIDTH_MM = 0.3
+
+REG_MARK_INFO_FONT_SIZE_PT = 2.5
+
+class InsertRegmark(EffectExtension):
+	def __init__(self):
+		EffectExtension.__init__(self)
+
+	def add_arguments(self, pars):
+		# Parse arguments
+		pars.add_argument("-X", "--reg-x", "--regwidth",  type = float, dest = "regwidth",   default = 180.0, help="X mark distance [mm]")
+		pars.add_argument("-Y", "--reg-y", "--reglength", type = float, dest = "reglength",  default = 230.0, help="Y mark distance [mm]")
+		pars.add_argument("--rego-x",  "--regoriginx",    type = float, dest = "regoriginx", default = 15.0,  help="X mark origin from left [mm]")
+		pars.add_argument("--rego-y", "--regoriginy",     type = float, dest = "regoriginy", default = 20.0,  help="X mark origin from top [mm]")
+		pars.add_argument("--verbose", dest = "verbose",  type = Boolean, default = False, help="enable log messages")
+
+	#SVG SVGd from (x,y) dimentional points
+	def points_to_svgd(self, p, capped=False):
+		f = p[0]
+		p = p[1:]
+		svgd = "M{:.5f},{:.5f}".format(f[0], f[1])
+		for x in p:
+			svgd += " L{:.5f},{:.5f}".format(x[0], x[1])
+		if capped:
+			svgd += "z"
+		return svgd
+
+	def effect(self):
+		reg_origin_X = self.options.regoriginx
+		reg_origin_Y = self.options.regoriginy
+		reg_width = self.options.regwidth if self.options.regwidth else int(self.svg.get("width").rstrip("mm")) - reg_origin_X*2
+		reg_length = self.options.reglength if self.options.reglength else int(self.svg.get("height").rstrip("mm")) - reg_origin_Y*2
+
+		if self.options.verbose == True:
+			self.msg(gettext("[INFO]: page width ")+str(self.svg.get("width").rstrip("mm")))
+			self.msg(gettext("[INFO]: page height ")+str(self.svg.get("height").rstrip("mm")))
+			self.msg(gettext("[INFO]: regmark from document left ")+str(reg_origin_X))
+			self.msg(gettext("[INFO]: regmark from document top ")+str(reg_origin_Y))
+			self.msg(gettext("[INFO]: regmark to regmark spacing X ")+str(reg_width))
+			self.msg(gettext("[INFO]: regmark to regmark spacing Y ")+str(reg_length))
+
+		# Register Mark #
+		mm_to_user_unit = self.svg.unittouu('1mm')
+
+		# Create a new register mark layer
+		regmark_layer = self.svg.add(Layer.new(REGMARK_LAYERNAME, id=REGMARK_LAYER_ID))
+		regmark_layer.transform = Transform(f"scale({mm_to_user_unit}, {mm_to_user_unit})")
+
+		# Create square in top left corner
+		regmark_layer.append(Rectangle.new(left=reg_origin_X, top=reg_origin_Y, width=REG_SQUARE_MM, height=REG_SQUARE_MM, id=REGMARK_TOP_LEFT_ID, style='fill:black;'))
+
+		# Create horizontal and vertical lines in group for top right corner
+		top_right_x = reg_origin_X+reg_width
+		top_right_path = [(top_right_x-REG_LINE_MM,reg_origin_Y), (top_right_x,reg_origin_Y), (top_right_x,reg_origin_Y + REG_LINE_MM)]
+		regmark_layer.append(PathElement.new(path=self.points_to_svgd(top_right_path), id=REGMARK_TOP_RIGHT_ID, style=f"fill:none; stroke:black; stroke-width:{REG_MARK_LINE_WIDTH_MM}"))
```

Comment:
you may reformulate:
```
regmark_layer.append(PathElement.new(path="M"+str(top_right_path), id=REGMARK_TOP_RIGHT_ID, style=f"fill:none; stroke:black; stroke-width:{REG_MARK_LINE_WIDTH_MM}"))

Reply:
How does this work? I thought `"M"+str([(1,2),(3,4)])` is `M[(1, 2), (3, 4)]`